User Onboarding

aseckin opened the pull request:
![image](https://github.com/user-attachments/assets/a434d013-8ebb-413a-a5cd-55b2fb4b38df)

[Live on play for testing: Create a new account and confirm your email](https://play.metaculus.com/?start_onboarding=true)

This adds a step-by-step wizard that guides the user through forecasting on various questions on Metaculus.

It starts immediately after email confirmation. For this, we have an `OnboardingCheck` component that triggers the Onboarding Modal.

Onboarding is an 'Immersive' modal that disables interaction with the background content. Base Modal component is updated with the `isImmersive` prop.

As user goes through the onboarding steps, their forecasts are submitted. I could use special attention to how these are being submitted, compared to how we're doing this everywhere else.

It also introduces a redesign to the Binary Slider component. It now has a Helper state which emphasizes the current position of the slider. It loses the ability to enter forecast via text input, but the new design performed much better in user interviews, so that was an intended compromise. 

Topics and questions used in this process are hardcoded in an `OnboardingSettings` file. This may not have perfect maintainability, but it should be fine to launch with. Would appreciate suggestions on how to make these editable via Django admin.

ncarazon reviewed (commented):
lgtm, but there is still a lot of linter errors and commented/unused code

elisescu commented on front_end/src/components/onboarding/steps/Step5.tsx:
```diff
@@ -0,0 +1,124 @@
+      forceNavigate(url);
+    } else {
+      console.log("No user found, force redirecting to /questions/");
+      forceNavigate("/questions/");
```
Can the onboarding really be triggered if no user? I would expect not to be the case

aseckin replied:
If the URL has ?start_onboarding=true it can be triggered. It's very unlikely, but could happen since this URL will be in their browser history

elisescu commented on front_end/src/components/onboarding/OnboardingSettings.ts:
```diff
@@ -0,0 +1,42 @@
+export const onboardingTopics = [
+  {
+    name: "US Politics",
```
What's the longer term plan for these? For now they are hard-coded in the code, but in a few weeks/months will become less relevant and will need to be updated. Do we want to move them to the backend to be changed in the admin, or?

aseckin replied:
Yes, the plan is to make these editable via admin. It shouldn't block launch, but the earlier we implement that would be better.

elisescu commented on front_end/src/app/(main)/questions/[id]/components/forecast_maker/binary_slider.tsx:
```diff
@@ -74,17 +88,48 @@ const BinarySlider: FC<Props> = ({
+        )} */}
       </div>
-      <div className="mb-3 block text-center">
+      {/* <div className="mb-3 block text-center">
```
The ForecastTextInpout seems to remain commented here. Not easy to see if this is intentional, but just flagging it to make sure it's not a mistake.

aseckin replied:
Yes, with the new design we lost the ability to enter text input for binary sliders.

aseckin commented on front_end/src/components/onboarding/steps/Step5.tsx:
```diff
@@ -0,0 +1,124 @@
+  const questionUrl = `/questions/${thirdQuestionId}`;
+
+  const forceNavigate = (url: string) => {
+    router.push(url);
+    setTimeout(() => {
+      window.location.href = url;
+    }, 100);
+  };
```
I tried using router.push, but I couldn't get it to work so switched to this method instead. I'll give it another shot

aseckin commented:
I've cleaned things up, fixed linting errors and removed console logs. This should be ready to merge now - I'll coordinate with Christian on the final content to launch with, and merge this.

aseckin commented:
@elisescu @ncarazon would be nice if you can take another look and confirm, as I'm planning to merge this some time tomorrow.
